### PR TITLE
added timezone in job scheduling

### LINKED ISSUE #172: Timezone support
Hi,

would it be possible to support an optional timezone setting in the job definition?

Regards,
Matteo

### PR DESCRIPTION
**Pull request by didiercrunch**
Added timezone field to jobs.  When timezon field is not set or set to the empty string, the local timezone will be used as in a backward compatible fashion.

Fixes #172

### CODE REVIEW

**Comment by victorcoder** [comment]
Thanks for the effort @didiercrunch I doubt if there is an enough benefit to introduce this code, it means more code to maintain and it's not clear to me that it will help a lot of use cases.

Dkron is meant to be run in servers and its a good practice to set server time to UTC and do the calculation yourself, it's not difficult and most programmers are already accustomed to do it because cron service doesn't have the possibility to set the TZ either.

**Comment by didiercrunch** [comment]
@victorcoder I totally understand what your concerns.  Code is a liability.  My own use case is to deal with local times in new york and london (which are impacted by summer daylight savings).  I'll work from my fork then.

cheers,

didier

**Comment by tlim-aspiration** [comment]
@didiercrunch Thanks for building this.  I might also use your branch as I have a requirement to launch my jobs at 8AM Pacific, no matter if it's daylight savings or standard.

**Comment by didiercrunch** [comment]
Neat!

**Comment by victorcoder** [comment]
After thinking again on this @didiercrunch, I'm going to accept this PR

See my comments.

File: cron/tz.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package cron
+
+import (
+	"time"
+)
+
+// Schedule wrapper with time zone awarness
+type timezoneAwareSchedule struct {
+	underlingSchedule Schedule
```
Comment (victorcoder) [inline cron/tz.go]: underlying?

**Comment by didiercrunch** [comment]
Cool!

On Feb 18, 2018 10:27 AM, "Victor Castell" <notifications@github.com> wrote:

After thinking again on this @didiercrunch <https://github.com/didiercrunch>,
I'm going to accept this PR

See my comments.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/victorcoder/dkron/pull/300#issuecomment-366523759>, or mute
the thread

File: cron/tz.go (hunk 1 above)
Comment (didiercrunch) [reply cron/tz.go]: totally, sorry I will fix that later on this week.  I am quite buzzy right now.

**Comment by giaplv57** [comment]
Hi all
The supporting timezone function is awesome, I suppose. Will help lots of case when we have Daylight Saving Time
Just wonder is this new function merged to the master branch? When will new version (with this function) release? 
Please update the wiki whenever new TZ is supported. 

Thanks,
Giap

**Comment by victorcoder** [comment]
@giaplv57 would be great to have it in the next release 0.10.0, let's see if @didiercrunch have time to give the final touches and I'll merge it.

**Comment by didiercrunch** [comment]
Ok, i'll work on it this week.

**Comment by digitalcrab** [comment]
Hello guys, thank you for the support. We also interested in this. Currently, we have 8 time zones where crons should be running. We do wired stuff to maintain it. This PR can save our life ;)

**Comment by victorcoder** [comment]
@didiercrunch could you solve conflict with v0.10.0?

**Pull request merged**